src/hotspot/share/runtime/thread.hpp

```diff
@@ -1014,6 +1014,11 @@ class JavaThread: public Thread {
   // Support for high precision, thread sensitive counters in JVMCI compiled code.
   jlong*    _jvmci_counters;
 
+  // Fast thread locals for use by JVMCI
+  intptr_t*  _jvmci_reserved0;
+  intptr_t*  _jvmci_reserved1;
+  oop        _jvmci_reserved_oop0;
```

Comment:
Can this use OopStorage? We've been getting rid of oop fields and the corresponding oops_do support.

Reply:
Wouldn't using OopStorage require an extra level of indirection for the field?

Reply:
It was https://bugs.openjdk.java.net/browse/JDK-8244997 - you were co-author :)

Maybe this jvmci_reserved_oop0 won't crash for the same reasons.  I don't know that.

I still would like to not see 45 lines of declarations for JVMCI added to JavaThread.  These should be in a separate header file and declared, as in https://bugs.openjdk.java.net/browse/JDK-8137018.  If you promise to fix 8137018, I'm fine with this change.

Reply:
co-author is a strong word.  :)  But that does ring a bell.  Why was threadObj problematic but the other existing oop fields were not?  JavaThread::oops_do_no_frames visits a lot of roots that aren't OopStorage.
I can tackle JDK-8137018.  I think we'll need to add an alias mechanism to vmStructs_jvmci.cpp to maintain backward compatibility but that's fairly straightforward.

Reply:
well, I didn't even list myself as author of that one.  _threadObj was a problem because some code (like thread dump management code) was accessing it from a terminating thread and the barriers were messed up. It was more complicated than that.  I don't know if this will be an issue for these declarations, and if you hide them in another place, we'll never know.

Reply:
I see.  These fields will only be accessed from generated code so I don't think there are the same runtime considerations with them.  Obviously it will be our problem to diagnose and fix if issues like that crop up.  We just don't want to break an obvious invariants that would require the use of OopStorage.

Reply:
Ok!